contracts/token/ERC20/utils/SafeERC20.sol

```diff
@@ -82,6 +82,51 @@ library SafeERC20 {
         }
     }
 
+    /**
+     * @dev Perform an {ERC1363} transferAndCall, with a fallback to the simple {ERC20} transfer if the target has no
+     * code. This can be used to implement an {ERC721}-like safe transfer that rely on {ERC1363} checks when
+     * targeting contracts.
+     */
+    function transferAndCallRelaxed(IERC1363 token, address to, uint256 value, bytes memory data) internal {
+        if (to.code.length == 0) {
+            safeTransfer(token, to, value);
+        } else {
+            token.transferAndCall(to, value, data);
```

Comment:
We should match the behavior of `safeTransfer`

```suggestion
            _callOptionalReturn(token, abi.encodeCall(token.transferAndCall, (to, value, data)));
```

Reply:
Well, I'm not so sure about that. ERC-20 didn't always have the boolean return, or at least some contract were deployed before it was implemented. That is a legacy we have to deal with. 
I don't think there is such a legacy for ERC-1363. If there are any contract that implement 1363 without returning a bool, I would consider them to not follow the standard and to be out of scope.

This change is not breaking anything, but its reducing the overall readability in order to support cases that should not exist, and that we should probably not support if they exist.

As to checking the return value. That is something we could do, but that the ERC doesn't really ask for, because the return of the function is described as 
```
   * @return true unless throwing
```

Reply:
Thanks for pointing out the ERC spec itself! I assumed it was just returning a boolean indicating the result.

You're right, and there shouldn't be legacy ERC1363 that justify this change. I also think those tokens would be non-standard and should be considered out of scope. However, the standard doesn't specify the return value meaning for `approveAndCall`.

I think we should check for the return value in `approveAndCall` given the ambiguity, what do you think?

Reply:
Would you be ok with the check being "high level"?

```solidity
    function transferAndCallRelaxed(IERC1363 token, address to, uint256 value, bytes memory data) internal {
        if (to.code.length == 0) {
            safeTransfer(token, to, value);
        } else {
            if (!token.transferAndCall(to, value, data)) {
                revert SafeERC20FailedOperation(token);
            }
        }
    }
```

Reply:
Note that `_callOptionalReturn` will pass if the call returns no data. IMO it should not.

Reply:
Yeah, I agree since it's completely non-standard to not return a value.
It works for me.